src/app/content/components/BookBanner.tsx

```diff
@@ -176,158 +176,137 @@ export const BarWrapper = styled.div<BarWrapperProps>`
   ${ifMiniNav(`margin-top: -${bookBannerDesktopMiniHeight}rem`)}
 `;
 
-export interface PropTypes {
-  pageNode?: ArchiveTreeSection;
-  book?: Book;
-  hasUnsavedHighlight?: boolean;
-  bookTheme: BookWithOSWebData['theme'];
-}
-
-interface BookBannerState {
-  scrollTransition: boolean;
-  tabbableBanner: 'mini' | 'big';
-}
-
 // tslint:disable-next-line:variable-name
-export class BookBanner extends Component<PropTypes, BookBannerState> {
-  public state: BookBannerState = {
-    scrollTransition: false,
-    tabbableBanner: 'big',
-  };
-  private miniBanner = React.createRef<HTMLDivElement>();
-  private bigBanner = React.createRef<HTMLDivElement>();
-
-  public handleScroll = () => {
-    if (this.miniBanner.current && this.bigBanner.current && typeof(window) !== 'undefined') {
-      const miniRect = this.miniBanner.current.getBoundingClientRect();
-      this.setState({
-        scrollTransition: miniRect.top === 0 &&
-          this.bigBanner.current.offsetTop + this.bigBanner.current.clientHeight > window.scrollY,
-        tabbableBanner: miniRect.top === 0 ? 'mini' : 'big',
-      });
+const BookBanner = () => {
+  const [scrollTransition, setScrollTransition] = React.useState(false);
+  const [tabbableBanner, setTabbableBanner] = React.useState('big');
+  const book = useSelector(select.book);
+  const pageNode = useSelector(select.pageNode);
+  const bookTheme = useSelector(select.bookTheme);
+  const hasUnsavedHighlight = useSelector(hasUnsavedHighlightSelector);
+  const miniBanner = React.useRef<HTMLDivElement>();
+  const bigBanner = React.useRef<HTMLDivElement>();
+  const services = useServices();
+
+  const handleScroll = () => {
+    if (miniBanner.current && bigBanner.current && typeof(window) !== 'undefined') {
+      const miniRect = miniBanner.current.getBoundingClientRect();
+
+      setScrollTransition(miniRect.top === 0
+        && bigBanner.current.offsetTop + bigBanner.current.clientHeight > window.scrollY);
+      setTabbableBanner(miniRect.top === 0 ? 'mini' : 'big');
```

Comment:
i think if we're going to be setting these at the same time and they're related we should put them in the same state